platforms/common/CMakeLists.txt

```diff
@@ -52,11 +52,14 @@ add_library(px4_platform
 	spi.cpp
 	${SRCS}
 )
-add_dependencies(px4_platform prebuild_targets)
+target_link_libraries(px4_platform prebuild_targets px4_work_queue)
+
+if("${PX4_BOARD}" MATCHES "sitl")
+	target_link_libraries(px4_platform drivers_board)
```

Comment:
this is ugly, but without this the "make tests" won't link properly. I wonder if there is a cleaner solution?